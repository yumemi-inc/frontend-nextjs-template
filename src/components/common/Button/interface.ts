/** @format */

export interface ButtonProps {
  example: string
}

export const ButtonMockProps: ButtonProps = {
  example: "hoge",
}
