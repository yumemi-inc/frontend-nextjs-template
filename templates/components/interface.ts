/** @format */

export interface TemplateNameProps {}

export const TemplateNameMockProps: TemplateNameProps = {}
