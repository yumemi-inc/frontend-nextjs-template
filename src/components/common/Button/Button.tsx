/** @format */

import React, { FC } from "react"
import { ButtonProps } from "./interface"

const Button: FC<ButtonProps> = () => <div data-testid="button">button Component</div>

export default Button
