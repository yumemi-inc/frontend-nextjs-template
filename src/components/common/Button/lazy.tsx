/** @format */
import React, { Suspense } from "react"
import dynamic from "next/dynamic"
import { ButtonProps } from "./interface"

const Button = dynamic(() => import("./Button"), { ssr: true })

const LazyButton = (props: ButtonProps) => (
  <Suspense fallback={null}>
    <Button {...props} />
  </Suspense>
)

export default LazyButton
