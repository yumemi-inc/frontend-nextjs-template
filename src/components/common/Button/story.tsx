/** @format */

import { Meta, StoryFn } from "@storybook/react"
import Button from "./Button"
import { ButtonProps } from "./interface"

export const Default = {}
export default {
  component: Button,
  argTypes: {},
} as Meta<typeof Button>

const Template: StoryFn<ButtonProps> = (props) => <Button {...props} />

export const Base = Template.bind({})
