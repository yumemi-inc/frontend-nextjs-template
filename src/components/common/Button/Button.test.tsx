/** @format */

import React from "react"
import { render, screen } from "@testing-library/react"
import "@testing-library/jest-dom/extend-expect"
import Button from "./Button"
import { ButtonMockProps } from "./interface"

describe("<Button />", () => {
  test("スナップショットテスト", () => {
    const { asFragment } = render(<Button {...ButtonMockProps} />)
    expect(asFragment()).toMatchSnapshot()
  })

  test('成功ケースのテスト', () => {
    render(<Button {...ButtonMockProps} />)
    const element = screen.getByTestId('button');
    expect(element.innerHTML).toBe('button Component');
  })

  test('失敗ケースのテスト', () => {
    render(<Button {...ButtonMockProps} />)
    const element = screen.getByTestId('button');
    expect(element.innerHTML).toBe('test');
  })
})
