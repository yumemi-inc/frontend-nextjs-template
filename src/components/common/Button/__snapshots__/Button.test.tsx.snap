// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> it should mount 1`] = `
<DocumentFragment>
  <div
    data-testid="button"
  >
    button Component
  </div>
</DocumentFragment>
`;

exports[`<Button /> スナップショットテスト 1`] = `
<DocumentFragment>
  <div
    data-testid="button"
  >
    button Component
  </div>
</DocumentFragment>
`;
